src/tmi/io_utils.py

```diff
@@ -0,0 +1,237 @@
+            for parent in reversed(pathlib.Path(path).parents):
+                if not os.path.exists(parent):
+                    raise FileNotFoundError(
+                        f"A bad path, {path}, was provided.\n"
+                        f"The folder, {parent.name}, could not be found..."
+                    )
+            raise FileNotFoundError(
+                f"The file/path, {pathlib.Path(path).name}, could not be found..."
+            )
+
+
+def list_files(dir_name, substrs=None, exact_match=False, ignore_hidden=True):
+    """List all files in a directory containing at least one given substring
+
+    Args:
+        dir_name (str):
+            Parent directory for files of interest
+        substrs (str or list):
+            Substring matching criteria, defaults to None (all files)
+        exact_match (bool):
+            If True, will match exact file names ('C' will match only 'C.tiff')
+            If False, will match substr pattern in file ('C' will match 'C.tiff' and 'CD30.tiff')
+        ignore_hidden (bool):
+            If True, will ignore hidden files. If False, will allow hidden files to be
+            matched against the search substring.
+
+    Returns:
+        list:
+            List of files containing at least one of the substrings
+    """
+    files = os.listdir(dir_name)
+    files = [file for file in files if not os.path.isdir(os.path.join(dir_name, file))]
+
+    # Filter out hidden files
+    if ignore_hidden:
+        files = [file for file in files if not file.startswith(".")]
+
+    # default to return all files
+    if substrs is None:
+        return files
+
+    # handle case where substrs is a single string (not wrapped in list)
+    if type(substrs) is not list:
+        substrs = [substrs]
+
+    if exact_match:
+        matches = [
+            file
+            for file in files
+            if any([substr == os.path.splitext(file)[0] for substr in substrs])
+        ]
+    else:
+        matches = [file for file in files if any([substr in file for substr in substrs])]
+
+    return matches
+
+
+def remove_file_extensions(files):
+    """Removes file extensions from a list of files
+
+    Args:
+        files (list):
+            List of files to remove file extensions from.
+            Any element that doesn't have an extension is left unchanged
+
+    Raises:
+        UserWarning:
+            Some of the processed file names still contain a period
+
+    Returns:
+        list:
+            List of files without file extensions
+    """
+
+    # make sure we don't try to split on a non-existent list
+    if files is None:
+        return
+
+    # remove the file extension
+    names = [os.path.splitext(name) for name in files]
+    names_corrected = []
+    extension_types = [
+        "tiff",
+        "tif",
+        "png",
+        "jpg",
+        "jpeg",
+        "tar",
+        "gz",
+        "csv",
+        "feather",
+        "bin",
+        "json",
+    ]
```

Comment:
Is it worth putting something like this in a `settings.py` file?

Reply:
Yeah, good call!